clang-tools-extra/clangd/Hover.cpp

```diff
@@ -138,15 +138,9 @@ std::string getNamespaceScope(const Decl *D) {
 
 std::string printDefinition(const Decl *D, PrintingPolicy PP,
                             const syntax::TokenBuffer &TB) {
-  if (auto *VD = llvm::dyn_cast<VarDecl>(D)) {
-    if (auto *IE = VD->getInit()) {
-      // Initializers might be huge and result in lots of memory allocations in
-      // some catostrophic cases. Such long lists are not useful in hover cards
-      // anyway.
-      if (200 < TB.expandedTokens(IE->getSourceRange()).size())
-        PP.SuppressInitializers = true;
-    }
-  }
+  // Initializers might be huge and result in lots of memory allocations in some
+  // catostrophic cases. Such long lists are not useful in hover cards anyway.
+  PP.EntireContentsOfLargeArray = false;
```

Comment:
i am not sure if this completely addresses the previous implementation. e.g. we can have a nested initializer list, which never has more than N elements directly, but the overall initalizer list might be huge. in such a scenario, we'll still end up allocating lots of strings.

so can we keep the old filtering while also setting this flag here? note that it'll still be ~incomplete. a more concrete approach could be based on counting number of "total" elements in the initializer list, and setting the suppression flag based on that. i'd actually lean towards such a solution, WDYT?

Reply:
> a more concrete approach could be based on counting number of "total" elements in the initializer list, and setting the suppression flag based on that. i'd actually lean towards such a solution, WDYT?

I agree, but don't know where to start on this. Since TokenBuffer doesn't have tokens from the preamble, what alternatives are there?

Reply:
I wasn't suggesting to use tokens, but rather count the number of elements directly through the AST.
we've got the expression available already, so we can go over its `chidlren()` and count total elements we **might** encounter, and suppress intializers based on that (when the tokens aren't available)

Reply:
OK, I check the size of the children() now when the initializer expression is not in the main file, and added two unit tests that verify that for a large initializer list from a header file it won't render, while a "small" initializer list will still render.